test/example/example.ts

```diff
@@ -1,52 +1,69 @@
 import {expect} from 'chai';
-import {createMockProvider, deployContract, getWallets, link} from '../../lib';
-import {Contract} from 'ethers';
+import {deployContract, link, createGanacheProvider, createBuidlerProvider, getWallets} from '../../lib';
+import {Contract, providers} from 'ethers';
 import BasicTokenMock from './build/BasicTokenMock.json';
 import MyLibrary from './build/MyLibrary.json';
 import LibraryConsumer from './build/LibraryConsumer.json';
 
+const copy = (obj: any) => JSON.parse(JSON.stringify(obj));
+
 describe('INTEGRATION: Example', () => {
-  const provider = createMockProvider();
-  const [wallet, walletTo] = getWallets(provider);
-  let token: Contract;
+  const providerCreators = [{
+    name: 'Ganache',
+    createProvider: createGanacheProvider
+  }, {
+    name: 'Buidler',
+    createProvider: createBuidlerProvider
+  }];
 
-  beforeEach(async () => {
-    token = await deployContract(wallet, BasicTokenMock, [wallet.address, 1000]);
-  });
+  let provider: providers.JsonRpcProvider;
 
-  it('Assigns initial balance', async () => {
-    expect(await token.balanceOf(wallet.address)).to.eq(1000);
-  });
+  providerCreators.forEach((providerCreator) => {
```

Comment:
Me Gusta 👍

Reply:
We should extract providers test setup to a separate file and use both providers in all tests related to providers, not only example (except fixtures ofc)